pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
Also read http://kubernetes.io/v1.0/docs/user-guide/managing-deployments.html

I don't want users to have to figure out whether to use patch or replace, how to construct a patch, whether to use --force or not, etc.

And we can't expect users to explicitly (esp. manually) specify every field they don't care about. That would break forward compatibility -- we wouldn't be able to add new fields to the API without forking a new API version.